packages/flutter/test/material/progress_indicator_test.dart

```diff
@@ -507,6 +507,27 @@ void main() {
     expect(find.byType(CircularProgressIndicator), paints..arc(strokeCap: StrokeCap.round));
   });
 
+  testWidgets('CircularProgressIndicator with custom animation controller', (
```

Comment:
I think we also need tests to verify the behavior when you change the controller, including from nonnull to null and null to nonnull.